tests/unit/test_results.py

```diff
@@ -0,0 +1,77 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from tmt.cli import TmtExitCode
+from tmt.result import ResultOutcome, results_to_exit_code
+
+
+@pytest.mark.parametrize(
+    ('outcomes', 'expected_exit_code'),
+    [
+        # "No test results found."
+        (
+            [],
+            TmtExitCode.NO_RESULTS_FOUND
+            ),
+        # "Errors occured during test execution."
+        (
+            [
+                ResultOutcome.PASS,
+                ResultOutcome.FAIL,
+                ResultOutcome.INFO,
+                ResultOutcome.SKIP,
+                ResultOutcome.WARN,
+                ResultOutcome.ERROR
+                ],
+            TmtExitCode.ERROR
+            ),
+        # "There was a fail or warn identified, but no error."
+        (
+            [
+                ResultOutcome.PASS,
+                ResultOutcome.FAIL,
+                ResultOutcome.INFO,
+                ResultOutcome.SKIP,
+                ResultOutcome.WARN,
+                ],
+            TmtExitCode.FAIL
+            ),
+        # "Test were executed, and all reported ``skip`` result."
+        (
+            [
+                ResultOutcome.SKIP,
+                ResultOutcome.SKIP,
+                ],
+            TmtExitCode.ALL_RESULTS_SKIPPED
```

Comment:
```suggestion
            TmtExitCode.ALL_TESTS_SKIPPED
```

It's not about results being skipped but tests.

Reply:
Fixed in b0cbb07c